mongocrypt/.gitignore

```diff
@@ -0,0 +1,39 @@
+*~
+.#*
+.git
+*#
+
+# os x stuff
+*Thumbs.db*
+*.DS_Store
+
+# Build artifacts
+build
+out
+
+# Eclipse files
+.classpath
+.project
+.settings
+
+# Intellij IDEA files
+*.ipr
+*.iws
+*.iml
+*.idea
+workspace.xml
+atlassian-ide-plugin.xml
+
+# gradle
+.gradle
+
+# local settings
+**/gradle.properties
+local.properties
```

Comment:
This is super minor, just for fun of tech discussion. In .gitignorr, "filrname" means the same as "**/filename" but the latter takes effect only after git v1.8.2

The only reason to use "**" is for such example: "a/**/b". 

No need to change if it has been a pattern elsewhere already. Feel free to ignore, without saying.

Reply:
Thanks for the insightful comment! We currently lack consistency in the root `.gitignore`. We could consider addressing this if it becomes a source of confusion.

As for now, I've removed the `.gitignore` from this module since we don’t need an additional one; there’s already a `.gitignore` in the root folder of our repo.